code/modules/research/ordnance/tank_compressor.dm

```diff
@@ -0,0 +1,330 @@
+				inserted_disk = attacking_disk
+			else
+				balloon_alert(user, span_warning("[attacking_disk] is stuck to your hand."))
+			return
+	if(!active && !inserted_tank)
+		if(default_deconstruction_screwdriver(user, "[base_icon_state]-open", "[base_icon_state]-open", item))
+			update_appearance()
+			return
+		if(panel_open)
+			if(default_deconstruction_crowbar(item))
+				return
+			if(default_change_direction_wrench(user, item))
+				return
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/tank_compressor/default_change_direction_wrench(mob/user, obj/item/wrench)
+	. = ..()
+	if(.)
+		/// Disconnect our partner.
+		if(nodes[1])
+			nodes[1].disconnect(src)
+			nodes[1] = null
+			if(parents[1])
+				nullify_pipenet(parents[1])
+		if(nodes[2])
+			nodes[2].disconnect(src)
+			nodes[2] = null
+			if(parents[2])
+				nullify_pipenet(parents[2])
+		set_init_directions()
+		/// Connect to a new one.
+		atmos_init()
+		if(nodes[1])
+			nodes[1].atmos_init()
+			nodes[1].add_member(src)
+		if(nodes[2])
+			nodes[2].atmos_init()
+			nodes[2].add_member(src)
+		SSair.add_to_rebuild_queue(src)
+		update_appearance()
+
+/// Glorified volume pump.
+/obj/machinery/atmospherics/components/binary/tank_compressor/process_atmos()
+	if(!active || !airs[2].total_moles() || !transfer_rate || !inserted_tank)
+		return
+
+	var/datum/gas_mixture/tank_air = inserted_tank.return_air()
+	if(airs[2].return_pressure() < 0.01 || tank_air.return_pressure() > TANK_COMPRESSOR_PRESSURE_LIMIT)
+		return
+
+	/// Prevent pumping if tank is taking damage but still below pressure limit. Here to prevent exploiting the buffer system.
+	if((inserted_tank.leaking) && (tank_air.return_pressure() <= TANK_LEAK_PRESSURE))
+		active = FALSE
+		return
+
+	var/datum/gas_mixture/removed = airs[2].remove_ratio(transfer_rate / airs[2].volume)
+	if(!removed)
+		return
+
+	tank_air.merge(removed)
+	update_parents()
+
+/obj/machinery/atmospherics/components/binary/tank_compressor/assume_air(datum/gas_mixture/giver)
+	if(!leaked_gas_buffer)
+		return ..()
+	leaked_gas_buffer.merge(giver)
+	return TRUE
+
+/// Unregistering of signals, clearing references of the tank, and recording of last pressure. Ran when a tank is about to explode or disintegrate.
+/obj/machinery/atmospherics/components/binary/tank_compressor/proc/unregister_tank()
+	SIGNAL_HANDLER
+	if(inserted_tank.get_integrity() <= 0)
+		flush_buffer()
+		var/datum/gas_mixture/tank_air = inserted_tank.return_air()
+		last_recorded_pressure = tank_air.return_pressure()
+		UnregisterSignal(inserted_tank, COMSIG_PARENT_QDELETING)
+		inserted_tank = null
+		active = FALSE
+		update_appearance()
+		return TRUE
+
+/// Use this to cancel explosions. Note that inserted_tank is not needed here and should already be unregistered by unregister_tank().
+/obj/machinery/atmospherics/components/binary/tank_compressor/proc/explosion_handle(atom/source, list/arguments)
+	SIGNAL_HANDLER
+	. = COMSIG_CANCEL_EXPLOSION
```

Comment:
why .?

Reply:
https://github.com/tgstation/tgstation/blob/11ccf197416a73bd3e187d7a035529e6112861af/code/controllers/subsystem/explosions.dm#L244-L245

Ill change it into a return too.